### PR TITLE
Add support for specifying database and table filter rules

### PR DESCRIPTION
**Pull request by suzaku**
### What problem does this PR solve? 

Add support for specifying database and table filter rules.

### What is changed and how it works?

1. Add filter related fields in `ChangeFeedDetail`
2. Add two methods for `ChangeFeedDetail` to check if a table should be ignored and filter a txn.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: cdc/model/changefeed.go [hunk 1]
```diff
@@ -31,6 +33,50 @@ type ChangeFeedDetail struct {
 	// The ChangeFeed will exits until sync to timestamp TargetTs
 	TargetTs uint64          `json:"target-ts"`
 	Info     *ChangeFeedInfo `json:"-"`
+
+	filter              *filter.Filter
+	FilterCaseSensitive bool          `json:"filter-case-sensitive"`
+	FilterRules         *filter.Rules `json:"filter-rules"`
```
Comment (amyangfei) [inline cdc/model/changefeed.go]: Is `black-white-list` better name here, which has been used in DM and lightning.

File: cdc/model/changefeed.go [hunk 2]
```diff
@@ -31,6 +33,50 @@ type ChangeFeedDetail struct {
...
+}
+
+func (detail *ChangeFeedDetail) getFilter() *filter.Filter {
+	if detail.filter == nil {
```
Comment (amyangfei) [inline cdc/model/changefeed.go]: I recommend not to combine system level filter with user-defined `black-white-list`.
In current design, If users want to ignore only one non-system level schema such as `not_replica_db`, they have to write `ignore-dbs: ["INFORMATION_SCHEMA", "PERFORMANCE_SCHEMA", "mysql", "not_replica_db"]` ?

File: cdc/model/changefeed.go (hunk 1 above)
Comment (amyangfei) [reply cdc/model/changefeed.go]: Besides,  how about adding a user-friendly way to add `black-white-list` config when they create a new changefeed. (Maybe add it in `cdc cli`, just for QA test temporarily)

File: cdc/model/changefeed_test.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
+// You may obtain a copy of the License at
```
Comment (amyangfei) [inline cdc/model/changefeed_test.go]: copyright is not complete

File: cdc/model/changefeed.go (hunk 1 above)
Comment (suzaku) [reply cdc/model/changefeed.go]: 1. The usage of `bwlist` seems confusing in DM, sometimes it references the filter itself and sometimes the rules, so I chose to reuse the name exposed by the filter package.
1. I can add that in a separate PR.

File: cdc/model/changefeed.go (hunk 2 above)
Comment (suzaku) [reply cdc/model/changefeed.go]: Fixed.

File: cdc/model/changefeed_test.go (hunk 3 above)
Comment (suzaku) [reply cdc/model/changefeed_test.go]: Fixed.

**Pull request merged**